docs/learning/backend/Workspace.md

```diff
@@ -1,475 +1,222 @@
-              "required": [
-                "name"
-              ]
-              "properties": {
-                "name": {
-                  "type": "string",
-                  "description": "template file name",
-                },
-                "loadByDefault: {
-                  "type": "boolean",
-                  "default": "true"
-                }
-              }
-            }
-          }
-        }
-      }
-    }
-  }
-}
+[[include:WorkspaceExamples.RegisterSchema]]
 ```
 
-### SettingDictionaries
+Your application should register its schemas shortly after invoking [IModelHost.startup]($backend). Registering a schema adds its typeDefs and settingDefs to [SettingsSchemas.typeDefs]($backend) and [SettingsSchemas.settingDefs]($backend), respectively. It also raises the [SettingsSchemas.onSchemaChanged]($backend) event. All schemas are unregistered when [IModelHost.shutdown]($backend) is invoked.
+
+## Settings dictionaries
 
-The *values* for one or more Settings may be established by creating a JavaScript object with properties matching [SettingName]($backend)s.
+The values of [Setting]($backend)s are provided by [SettingsDictionary]($backend)s. The [Settings]($backend) for the current session can be accessed via the `settings` property of [IModelHost.appWorkspace]($backend). You can add new dictionaries to provide settings values at any time during the session, although most dictionaries will be loaded shortly after [IModelHost.startup]($backend).
 
-E.g.:
+Let's load a settings dictionary that provides values for some of the settings in the LandscapePro schema:
 
 ```ts
-[[include:Settings.addDictionaryDefine]]
+[[include:WorkspaceExamples.AddDictionary]]
 ```
 
-> Note: The types of the properties should match the `SettingType` declared in the SettingGroupSec.
-
-Then, the dictionary can be given a [DictionaryName]($backend), and a [SettingsPriority]($backend) and be added to the current [Settings]($backend):
+Now you can access the setting values defined in the dictionary via `IModelHost.appWorkspace.settings`:
 
 ```ts
-[[include:Settings.addDictionary]]
+[[include:WorkspaceExamples.GetSettings]]
 ```
 
-Values in `SettingDictionary`s with a higher `SettingsPriority` override values in dictionaries with a lower priority.
+Note that `getString` returns `undefined` for "landscapePro/preferredStyle" because our dictionary didn't provide a value for it. The overload of that function (and similar functions like [Settings.getBoolean]($backend) and [Settings.getObject]($backend)) allows you to specify a default value to use if the value is not defined.
+
+> Note: In general, avoid caching the values of individual settings - just query them each time you need them, because they can change at any time. If you must cache (for example, if you are populating a user interface from the setting values), listen for and react to the [Settings.onSettingsChanged]($backend) event.
```

Comment:
I understand not caching settings. What about caching Workspace resources? 

In the example, would you expect the `getAvailableTrees` function to be called every time someone ran/installed the `place-dogwood` or `place-shrub` tool?

Reply:
Yes, those resources are accessed via settings that point to the relevant workspace dbs.

Reply:
Added a paragraph to the end after the `getAvailableTrees` example.